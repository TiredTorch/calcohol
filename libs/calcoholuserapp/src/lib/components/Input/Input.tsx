import { FC } from "react";
import { TextField as MUIInput, TextFieldProps } from "@mui/material";

export const Input: FC<TextFieldProps> = ({
	...rest
}) => {
	return (
		<MUIInput
			variant="standard"
			InputProps={{
				disableUnderline: true
			}}
			{...rest}
		/>
	);
};

export default Input;