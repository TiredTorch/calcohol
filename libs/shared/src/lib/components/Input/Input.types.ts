import { InputProps as MUIInputPtops } from "@mui/material";

export type InputProps = MUIInputPtops & {
	error: boolean | undefined;
	helperText: string | false | undefined;
};