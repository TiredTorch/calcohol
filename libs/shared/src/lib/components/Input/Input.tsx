import { FC } from "react";
import { Box, FormHelperText, Input as MUIInput } from "@mui/material";
import { InputProps } from "./Input.types";

export const Input: FC<InputProps> = ({
	error,
	helperText,
	...rest
}) => {
	return (
		<Box>
			<MUIInput
				disableUnderline
				{...rest}
			/>
			<FormHelperText
				error={error}
			>
				{helperText}
			</FormHelperText>
		</Box>
	);
};

export default Input;