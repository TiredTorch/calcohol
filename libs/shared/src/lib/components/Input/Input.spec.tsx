import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { faker } from "@faker-js/faker";
import { Input } from "./Input";

afterEach(() => {
	jest.resetAllMocks();
});

const errorHelperClassName = "Mui-error";
const mockedHelperText = faker.datatype.string(10);

it("should render input without error", () => {
	const { baseElement } = render(
		<Input
			error={false}
			helperText={mockedHelperText}
		/>
	);

	expect(baseElement).toBeInTheDocument();
	expect(baseElement).toMatchSnapshot();
});

it("should render input with error", () => {
	const { baseElement } = render(
		<Input
			error
			helperText={mockedHelperText}
		/>
	);

	expect(baseElement).toBeInTheDocument();
	expect(baseElement).toMatchSnapshot();
	expect(baseElement.getElementsByClassName(errorHelperClassName)[0]).toBeInTheDocument();
});