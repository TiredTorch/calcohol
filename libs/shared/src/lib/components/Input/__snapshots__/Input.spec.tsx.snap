// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render input with error 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiInputBase-root MuiInput-root MuiInputBase-colorPrimary css-spqcwk-MuiInputBase-root-MuiInput-root"
      >
        <input
          class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          type="text"
          value=""
        />
      </div>
      <p
        class="MuiFormHelperText-root Mui-error css-1d1r5q-MuiFormHelperText-root"
      >
        {!?x2UtT^d
      </p>
    </div>
  </div>
</body>
`;

exports[`should render input without error 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiInputBase-root MuiInput-root MuiInputBase-colorPrimary css-spqcwk-MuiInputBase-root-MuiInput-root"
      >
        <input
          class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          type="text"
          value=""
        />
      </div>
      <p
        class="MuiFormHelperText-root css-1d1r5q-MuiFormHelperText-root"
      >
        {!?x2UtT^d
      </p>
    </div>
  </div>
</body>
`;
